activerecord/lib/active_record/migration.rb

```diff
@@ -613,6 +613,10 @@ def migrate(direction)
       def disable_ddl_transaction!
         @disable_ddl_transaction = true
       end
+
+      def connection
+        ActiveRecord::Base.connection
```

Comment:
Could we pass this in at the construction site rather than just hiding the dependency in a method?